options/logger.go

```diff
@@ -82,190 +56,171 @@ func (f LogFormat) MakeFormatter() (send.MessageFormatter, error) {
 	}
 }
 
-// Buffer packages options for whether or not a Logger should be
-// buffered and the duration and size of the respective buffer in the case that
-// it should be.
-type Buffer struct {
-	Buffered bool          `bson:"buffered" json:"buffered" yaml:"buffered"`
-	Duration time.Duration `bson:"duration" json:"duration" yaml:"duration"`
-	MaxSize  int           `bson:"max_size" json:"max_size" yaml:"max_size"`
+// LogConfig type aliases []byte in order to implement the bson.MarshalBSON and
+// json.MarshalJSON interfaces. It is intended to hold the logger configuration
+// data.
+type LogConfig []byte
+
+func (lc LogConfig) MarshalBSON() ([]byte, error) { return lc, nil }
+func (lc LogConfig) MarshalJSON() ([]byte, error) { return lc, nil }
+
+// LoggerConfig represents the necessary information to construct a new grip
+// send.Sender. LoggerConfig implements the bson.MarshalBSON and
+// json.MarshalJSON interfaces.
+type LoggerConfig struct {
+	Type     string          `json:"type" bson:"type"`
+	Format   LogConfigFormat `json:"format" bson:"format"`
+	Config   LogConfig       `json:"config" bson:"config"`
+	Registry LoggerRegistry  `json:"-" bson:"-"`
+
+	producer LoggerProducer
+	sender   send.Sender
 }
 
-// Validate ensures that BufferOptions is valid.
-func (opts Buffer) Validate() error {
-	if opts.Buffered && opts.Duration < 0 || opts.MaxSize < 0 {
-		return errors.New("cannot have negative buffer duration or size")
+// Validate ensures the LoggerConfig is valid.
+func (lc *LoggerConfig) Validate() error {
+	catcher := grip.NewBasicCatcher()
+
+	catcher.NewWhen(lc.Type == "", "cannot have empty logger type")
+	catcher.Add(lc.Format.Validate())
+
+	if lc.Registry == nil {
+		lc.Registry = globalLoggerRegistry
 	}
-	return nil
+
+	return catcher.Resolve()
 }
 
-// Log contains options related to the logging done by Jasper.
-//
-// By default, logger reads from both standard output and standard error.
-type Log struct {
-	BufferOptions        Buffer                    `json:"buffer_options,omitempty" bson:"buffer_options"`
-	BuildloggerOptions   send.BuildloggerConfig    `json:"buildlogger_options,omitempty" bson:"buildlogger_options"`
-	BuildloggerV3Options timber.LoggerOptions      `json:"buildlogger_v3_options" bson:"buildlogger_v_3_options"`
```

Comment:
Since the buildlogger v3 isn't implemented anymore with timber, is the expectation that we're removing the vendored timber so external packages (i.e. evergreen) implement it?

If so, can we remove the timber references from `make vendor-clean` and remove the vendor directory?

Reply:
yes!